src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -137,8 +134,8 @@ private static void AddStandbyServices(this IServiceCollection services)
             // Core script host service
             services.AddSingleton<IHostedService>(p =>
             {
-                var hostEnvironment = p.GetService<IScriptWebHostEnvironment>();
-                if (hostEnvironment.InStandbyMode)
+                var appHostOptions = p.GetService<IOptionsMonitor<ScriptApplicationHostOptions>>();
+                if (appHostOptions.CurrentValue.InStandbyMode)
```

Comment:
When should this new value be used v.s. IScriptWebHostEnvironment.InStandbyMode? We still use the latter all over the place. We need clear guidance on when one or the other should be used.

Reply:
Not only that... we use `IEnvironment.IsPlaceholderModeEnabled()` in a bunch of places too. We need to do a sweep through here and make sure we're only checking this directly against the environment in very few places. I've filed #4368 to track this work.